### PR TITLE
Added ability to upload books in english language

### PR DESCRIPTION
**Pull request by Anshul-Birla**
### Administrative Info
Monday Board ID: 853642866
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added uploading book wizard 
- Added ckeditor to project 
- changed image route in backend 

### Testing
How did you confirm your changes worked? 
- Uploaded books with different images and different markdown in body to make sure everything worked

### Confirmation of Change 
- Navigate to the upload books page,  click new book,  add the details of the book, and upon submit, see if the new book pops up on the upload books page

### CODE REVIEW

**Review by typokign — CHANGES_REQUESTED** [review]
Looks great!! A few things to start with:

* The New Book button is rendering like a link (underlined). Since it's already a button, I think it would be better to add an `onClick={() => useHistory().push('/books/new')}` to it.
* Can we add 16px of left/right padding on the form fields so they're a little less crowded?
* The final Upload Book confirm button is missing some styling
* You should be able to click the green circles to switch between tabs, but only if all of the previous tabs are valid.
* We're missing the confirmation modal when you click the Upload Book button.
* The "Coming Soon!" text at the end isn't very informative IMO. Maybe we should say "Mobile preview coming soon. Please double-check all screens before submitting!" or something of that effect?

Great work once again, this really came together fast!!

File: web/src/pages/BookWizard/GeneralPage.module.css [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    justify-content: center;
+}
+
+.dropZoneText {
+    composes: h3 from '../../App.css'
+}
+
+.dropZoneImage {
+    width: 100%;
+    height: 100%;
```
Comment (dacruz21) [inline web/src/pages/BookWizard/GeneralPage.module.css:33]: Can you add `object-fit: cover;` here? Right now if you upload a non-square image, it stretches the image to fit the box. In the app, the image is cropped to fit the box, so we should match that here.

File: web/src/pages/BookWizardPage.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  
+  // controls wheter to redirect the page back to start
+  const [redirect, setRedirect] = useState<boolean>(false);
+
+  const pages = [
+    <GeneralPage key={0} onTitleChange={setTitle} onAuthorChange={setAuthor} onImageChange={setImage} currentImage={image} currentTitle={title} currentAuthor={author}></GeneralPage>,
+    <TabContentPage onContentChange={setReadTabContent} key={1} currentContent={readTabContent}></TabContentPage>,
+    <TabContentPage onContentChange={setExploreTabContent} currentContent={exploreTabContent} key={2}></TabContentPage>,
+    <TabContentPage onContentChange={setLearnTabContent} currentContent={learnTabContent} key={3}></TabContentPage>,
+    <OverviewPage onSubmit={submitPage} key={4}></OverviewPage>
```
[the remark this answers is not in the document]
Comment (Anshul-Birla) [reply web/src/pages/BookWizardPage.tsx:69]: it will not compile if i dont add keys lol

**Review by typokign — DISMISSED** [review]
Great work, the page select bubbles are really intuitive!

File: web/src/pages/BookWizard/GeneralPage.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    if(acceptableMimeType.indexOf(imageFile.type) == -1) {
+      alert("Please enter a jpeg or a png file");
+      return;
+    } 
+    else {
+      handleFile(imageFile);
+    }
+  };
+
+  const handleFile = (file: File): void => {
```
Comment (aksharans) [inline web/src/pages/BookWizard/GeneralPage.tsx:61]: Uploading an image, then clicking the image to upload again, then clicking cancel gives this error:
> TypeError: Failed to execute 'createObjectURL' on 'URL': Overload resolution failed.

Could a fix just be to add a null check? :
```
    if (file != null){ 
      onImageChange(file);   
      setPreviewUrl(URL.createObjectURL(file));   
    }  
```

File: web/src/pages/BookWizard/TabContentPage.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        </div>
+      
+        <input 
+          type="text"
+          className={styles.inputField}
+          value={video}
+          onChange={ e => setVideo(e.target.value)}
+        />
+        <div className = {styles.editorText}>
+          Write Here
```
Comment (aksharans) [inline web/src/pages/BookWizard/TabContentPage.tsx:64]: A style thing I noticed: though it wasn't specified in the mockup/style guide, would a red asterisk for required field be added here like in the title/author/image?

File: web/src/pages/BookWizard/GeneralPage.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    onImageChange(file);
+    setPreviewUrl(URL.createObjectURL(file));
+  };
```
Comment (aksharans) [inline web/src/pages/BookWizard/GeneralPage.tsx:64]: Uploading an image, then clicking the image to upload again, then clicking cancel gives this error:
> TypeError: Failed to execute 'createObjectURL' on 'URL': Overload resolution failed.

Could a fix just be to add a null check? :
```
    if (file != null){ 
      onImageChange(file);   
      setPreviewUrl(URL.createObjectURL(file));   
    }  
```

File: web/src/pages/BookWizard/GeneralPage.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    e.preventDefault();
+  };
+
+  const handleOnDrop = (e: React.DragEvent): void => {
+    e.preventDefault();
+    e.stopPropagation();
+    const imageFile = e.dataTransfer.files[0];
+    if(acceptableMimeType.indexOf(imageFile.type) == -1) {
+      alert("Please enter a jpeg or a png file");
+      return;
```
Comment (aksharans) [inline web/src/pages/BookWizard/GeneralPage.tsx:54]: After uploading an image, then clicking on that same image and dragging/dropping it within the input, gives this error:
> TypeError: Cannot read property 'type' of undefined

Could a similar fix be :
```
    if(imageFile != null && acceptableMimeType.indexOf(imageFile.type) == -1) {
      alert("Please enter a jpeg or a png file");
      return;
    } 
```

**Review by aksharans — CHANGES_REQUESTED** [review]
Looks really good & works really well. Just a couple of things I noticed involving the image upload and a style suggestion.
Also, I was getting linting warnings but I'm not sure if they were expected or not:
```
src/pages/BookWizard/TabContentPage.tsx
  Line 38:41:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  Line 38:48:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  Line 39:76:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  Line 70:31:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  Line 70:44:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
```

File: web/src/pages/BookWizard/TabContentPage.tsx (hunk 4 above)
Comment (Anshul-Birla) [reply web/src/pages/BookWizard/TabContentPage.tsx:64]: yeah did not add it cuz it wasnt in the style guide. Should I add it @dacruz21

File: web/src/pages/BookWizard/GeneralPage.tsx (hunk 5 above)
Comment (Anshul-Birla) [reply web/src/pages/BookWizard/GeneralPage.tsx:64]: good catch! Fixed both errors you pointed out :)

**Comment by Anshul-Birla** [comment]
> Looks really good & works really well. Just a couple of things I noticed involving the image upload and a style suggestion.
> Also, I was getting linting warnings but I'm not sure if they were expected or not:
> 
> ```
> src/pages/BookWizard/TabContentPage.tsx
>   Line 38:41:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
>   Line 38:48:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
>   Line 39:76:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
>   Line 70:31:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
>   Line 70:44:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
> ```
I think this is ok. Ckeditor doesn't have types, so had to use any for them

File: web/src/pages/BookWizard/TabContentPage.tsx (hunk 4 above)
Comment (dacruz21) [reply web/src/pages/BookWizard/TabContentPage.tsx:64]: Sure, not a bad idea.

Comment (Anshul-Birla) [reply web/src/pages/BookWizard/TabContentPage.tsx:65]: done :0

**Review by aksharans — COMMENTED** [review]
I was still getting an error when dragging and dropping an existing image to the input box, so I think you may need to modify line 54 in GeneralPage.tsx to 
```
if(imageFile != null && acceptableMimeType.indexOf(imageFile.type) == -1) {
```
but that's just an edge case. Otherwise, LGTM

**Review by typokign — APPROVED** [review]

**Review by aksharans — APPROVED** [review]

**Pull request merged**